src/parser/decompiler/decompile.sh

```diff
@@ -8,13 +8,25 @@ FILES=("scripts/heroes" "scripts/abilities" "scripts/generic_data" "scripts/misc
 
 cp "$DEADLOCK_PATH/game/citadel/steam.inf" "decompiled-data/version.txt"
 
-# Loop through files and run Decompiler.exe for each
+ # Detect the operating system
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+      # MacOS
+      DECOMPILER_EXEC="Decompiler"  # Mac executable without .exe
+  elif [[ "$OSTYPE" == "msys" || "$OSTYPE" == "cygwin" ]]; then
+      # Windows (via Git Bash or Cygwin)
+      DECOMPILER_EXEC="Decompiler.exe"  # Windows executable with .exe
+  else
+      echo "Unsupported operating system."
+      exit 1
+  fi
+
+# Loop through files and run Decompiler for each
 for FILE in "${FILES[@]}"; do
   INPUT_PATH="$DEADLOCK_PATH/game/citadel/pak01_dir.vpk"
   VPK_FILEPATH="${FILE}.vdata_c"
-  
-  # Run the decompiler
-  "$DECOMPILER_PATH/Decompiler.exe" -i "$INPUT_PATH" --output "$OUTPUT_DIR" --vpk_filepath "$VPK_FILEPATH" -d
+
+# Run the decompiler
+"$DECOMPILER_PATH/$DECOMPILER_EXEC" -i "$INPUT_PATH" --output "$OUTPUT_DIR" --vpk_filepath "$VPK_FILEPATH" -d
```

Comment:
Could we instead just have `DECOMPILER_PATH` variable include the file name on the end? That means no need for any fancy OS detection

Reply:
Good call 👍. Ill probably get to it tomorrow